deployment/cre/jobs/operations/propose_gateway_job.go

```diff
@@ -51,140 +51,170 @@ var ProposeGatewayJob = operations.NewOperation[ProposeGatewayJobInput, ProposeG
 	"propose-gateway-job-op",
 	semver.MustParse("1.0.0"),
 	"Propose Gateway Job",
-	func(b operations.Bundle, deps ProposeGatewayJobDeps, input ProposeGatewayJobInput) (ProposeGatewayJobOutput, error) {
-		targetDONs := make([]pkg.TargetDON, 0)
-		for _, ad := range input.DONs {
-			filter := offchain.TargetDONFilter{
-				Key:   offchain.FilterKeyDONName,
-				Value: ad.Name,
-			}
-			nodes, err := pkg.FetchNodeChainConfigsFromJD(deps.Env.GetContext(), deps.Env, filter)
-			if err != nil {
-				return ProposeGatewayJobOutput{}, err
-			}
-
-			fam, chainID, err := parseSelector(uint64(input.GatewayKeyChainSelector))
-			if err != nil {
-				return ProposeGatewayJobOutput{}, err
-			}
-
-			req := pkg.FetchNodesRequest{
-				Domain:  input.Domain,
-				Filters: []offchain.TargetDONFilter{filter},
-			}
-			ns, err := pkg.FetchNodesFromJD(deps.Env.GetContext(), deps.Env, req)
-			if err != nil {
-				return ProposeGatewayJobOutput{}, err
-			}
-
-			// make map of node id to node
-			m := make(map[string]*nodev1.Node, len(ns))
-			for _, n := range ns {
-				m[n.Id] = n
-			}
-
-			var members []pkg.TargetDONMember
-			for _, n := range nodes {
-				var found bool
-				for _, cc := range n.ChainConfigs {
-					if cc.Chain.Id == chainID && cc.Chain.Type == fam {
-						nodeName := n.NodeID
-						if matched, ok := m[n.NodeID]; ok {
-							nodeName = matched.Name
-						}
-						members = append(members, pkg.TargetDONMember{
-							Address: cc.AccountAddress,
-							Name:    fmt.Sprintf("%s (DON %s)", nodeName, ad.Name),
-						})
-						found = true
-
-						break
-					}
-				}
+	proposeGatewayJob,
+)
 
-				if !found {
-					return ProposeGatewayJobOutput{}, fmt.Errorf("could not find key belonging to chain id %s", chainID)
-				}
-			}
+func proposeGatewayJob(b operations.Bundle, deps ProposeGatewayJobDeps, input ProposeGatewayJobInput) (ProposeGatewayJobOutput, error) {
```

Comment:
_nit_ some docstring on what nodes are needed to make a gateway job

Reply:
What nodes are needed? I don't follow